### PR TITLE
drop type specs in RecoEgamma

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description: 
Update the safer syntax for existing parameter :
- drop type specifications where the original parameter exists.
- move all parameter inside the clone

Instead of modifying parameters with full type specs, which can be interpreted as an insertion of a new parameter, it is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.
(The references were [PR#30700](https://github.com/cms-sw/cmssw/pull/30700), [PR#30827](https://github.com/cms-sw/cmssw/pull/30827), [PR#30947](https://github.com/cms-sw/cmssw/pull/30947),[PR#31162](https://github.com/cms-sw/cmssw/pull/31162),)

In this PR, total 36 files updated. 
- RecoEgamma/{EgammaPhotonProducers} 23 files 
- RecoEgamma/{EgammaElectronProducers} 5 files 
- RecoEgamma/{EgammaIsolationAlgos} 4 files 
- RecoEgamma/{EgammaTools} 2 files 
- RecoEgamma/{ElectronIdentification} 1 file 
- RecoEgamma/{PhotonIdentification} 1 file 

#### PR validation:
Event Content comparison check was also done and there is no change with these updates.
Tested in CMSSW_11_2_X, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

File: RecoEgamma/EgammaElectronProducers/python/electronPFIsolationValues_cff.py [hunk 1]
```diff
@@ -72,21 +72,21 @@
 )
 
 elPFIsoValueCharged04 = elPFIsoValueCharged03.clone()
-elPFIsoValueCharged04.deposits[0].deltaR = cms.double(0.4)
+elPFIsoValueCharged04.deposits[0].deltaR = 0.4
```
Comment (slava77) [inline RecoEgamma/EgammaElectronProducers/python/electronPFIsolationValues_cff.py:75]: I was looking at a similar case earlier today, it looks like the following syntax works and can let us further simplify the replacement cases in vectors
```suggestion
elPFIsoValueCharged04 = elPFIsoValueCharged03.clone(deposits = {0: dict(deltaR = 0.4)} )
```

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
This PR ends up with several additional warnings in the logs, for example:
```
%MSG-w Configuration:   ConversionTrackMerger:generalConversionStepConversionTrackMerger@ctor  26-Aug-2020 14:19:56 CEST pre-events
Warning:
	string TrackProducer1 = "generalConversionTrackProducer"
is deprecated, please update your config file to use
	InputTag TrackProducer1 = generalConversionTrackProducer
%MSG
%MSG-w Configuration:   ConversionTrackMerger:generalConversionStepConversionTrackMerger@ctor  26-Aug-2020 14:19:56 CEST pre-events
Warning:
	string TrackProducer2 = "conversionStepConversionTrackProducer"
is deprecated, please update your config file to use
	InputTag TrackProducer2 = conversionStepConversionTrackProducer
%MSG
```

This is due to those two parameters `TrackProducer1` and `TrackProducer2` which are wrongly defined as "string" in the file `RecoEgamma/EgammaPhotonProducers/python/conversionTrackMerger_cfi.py`. Without this PR the two parameters were re-defined as `InputTag`, while with this PR they derive the same (wrong) `string` definition of the cloned config.

To fix it, please redefine
```PYTHON
    TrackProducer1 = cms.string(''),
    TrackProducer2 = cms.string(''),
```
as
```PYTHON
    TrackProducer1 = cms.InputTag(''),
    TrackProducer2 = cms.InputTag(''),
```
in `RecoEgamma/EgammaPhotonProducers/python/conversionTrackMerger_cfi.py`

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
> * No significant changes to the logs found

Ok, better now

**Comment by perrotta** [comment]
+1
- Technical cleaning of the syntax for parameters passed to cloned/modified modules, as specified in the PR description
- Jenkins tests pass

**Comment by silviodonato** [comment]
+1

**Pull request merged**